### PR TITLE
Map registry

### PR DESCRIPTION
**Pull request by EtienneDesticourt**

### CODE REVIEW

File: lib/entice/logic/maps/map.ex [hunk 1]
```diff
@@ -52,7 +52,8 @@ defmodule Entice.Logic.Map do
 
 
   defp content(mod) do
-    name = mod |> Module.split |> List.last |> to_string
+    [map_atom] = elem(mod,2)
```
Comment (ephe-meral) [inline lib/entice/logic/maps/map.ex]: i'm not sure if i understand whats happening here... what do we get passed in?

File: lib/entice/logic/map_registry.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
+defmodule Entice.Logic.MapRegistry do
+  @doc """
+  Stores all the instances for each map.
+  Its state is in the following format: %{map=>entity_id}
+  """
+  use GenServer
```
Comment (ephe-meral) [inline lib/entice/logic/map_registry.ex]: not necessary if you're using an agent

File: lib/entice/logic/map_registry.ex [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      _ ->
+        {:error, :instance_already_running}
+    end
+  end
+
+  def get_instance(map),
+  do: Agent.get(__MODULE__, fn state -> state |> Map.get(map) end)
+
+  def instance_stopped(map),
+  do: Agent.cast(__MODULE__, &remove_instance(&1, map))
```
Comment (ephe-meral) [inline lib/entice/logic/map_registry.ex]: i'd say we can use an inline function here, just as above, no?

File: lib/entice/logic/maps/map.ex (hunk 1 above)
Comment (EtienneDesticourt) [inline lib/entice/logic/maps/map.ex]: We get the unquoted mapname as far as I understand which looks something like [something, something, [:Lobby]].
So we get element at index 2, [:Lobby] and extract the atom from the list.

**Comment by ephe-meral** [comment]
:heart:

**Pull request merged**